nni/algorithms/compression/tensorflow/pruning/one_shot.py

```diff
@@ -57,15 +62,54 @@ def calc_masks(self, sparsity, wrapper, wrapper_idx=None):
 
             w_abs = tf.math.abs(weight)
             k = tf.size(weight) - num_prune
-            topk = tf.math.top_k(tf.reshape(w_abs, [-1]), k)[0]
+            topk = tf.math.top_k(tf.reshape(w_abs, [-1]), k).values
             if tf.size(topk) == 0:
                 mask = tf.zeros_like(weight)
             else:
                 mask = tf.math.greater_equal(w_abs, topk[-1])
             masks[weight_variable.name] = tf.cast(mask, weight.dtype)
         return masks
 
+class SlimPrunerMasker(WeightMasker):
+    def __init__(self, model, pruner, **kwargs):
+        super().__init__(model, pruner)
+        weight_list = []
+        for wrapper in pruner.wrappers:
+            weights = [w for w in wrapper.layer.weights if '/gamma:' in w.name]
+            assert len(weights) == 1, f'Bad weights: {[w.name for w in wrapper.layer.weights]}'
+            weight_list.append(tf.math.abs(weights[0].read_value()))
+        all_bn_weights = tf.concat(weight_list, 0)
+        k = int(all_bn_weights.shape[0] * (1 - pruner.wrappers[0].config['sparsity']))
+        top_k = tf.math.top_k(tf.reshape(all_bn_weights, [-1]), k).values
+        self.global_threshold = top_k.numpy()[-1]
 
-MASKER_DICT = {
-    'level': LevelPrunerMasker,
-}
+    def calc_masks(self, sparsity, wrapper, wrapper_idx=None):
+        assert isinstance(wrapper.layer, tf.keras.layers.BatchNormalization), \
+                'SlimPruner only supports 2D batch normalization layer pruning'
+
+        weight = None
+        weight_name = None
+        bias_name = None
+
+        for variable in wrapper.layer.weights:
+            if '/gamma:' in variable.name:
+                weight = variable.read_value()
+                weight_name = variable.name
+            elif '/beta:' in variable.name:
+                bias_name = variable.name
+
+        assert weight is not None
+        if wrapper.masks.get(weight_name) is not None:
+            weight *= wrapper.masks[weight_name]
+
+        filters = weight.shape[0]
+        num_prune = int(filters * sparsity)
+        if filters >= 2 and num_prune >= 1:
```

Comment:
Does it contain some corner cases? Why not casting weight by `self.global_threshold` directly instead of doing such branch identification.